skimage/feature/blob.py

```diff
@@ -295,19 +323,24 @@ def blob_log(image, min_sigma=1, max_sigma=50, num_sigma=10, threshold=.2,
     # computing gaussian laplace
     # s**2 provides scale invariance
     gl_images = [-gaussian_laplace(image, s) * s ** 2 for s in sigma_list]
-    image_cube = np.dstack(gl_images)
+
+    # Replace by image_cube = np.stack(hessian_images, axis=-1)
+    # When we upgrade minimal requirements to NumPy 1.10
+    sl = (slice(None),) * image.ndim + (np.newaxis,)
+    arrays = [np.asanyarray(arr) for arr in gl_images]
+    extended_arrays = [arr[sl] for arr in arrays]
+    image_cube = np.concatenate(extended_arrays, axis=-1)
```

Comment:
Actually I just saw that these arrays are just used internally, so restacking wouldn't change the API... So I still think you should update. It's an antipattern to stack in this order in NumPy, so it's best if we avoid it in our own code! I think it simplifies the code a bit, too, to stack along first axis.

Plus you get to close the [issue](https://github.com/scikit-image/scikit-image/issues/2173) I just opened. =P